### ISSUE TITLE
Output array sum ist not 1 / Action Recognition on own Dataset

### ISSUE THREAD

**Issue by yh80ywol** [dmlc/gluon-cv#1464]
Hello,

i trained a action recognition model on my own dataset.
It works perfect, but i´m confused about the output arrays.
Shouldn´t they vary between 0 and 1 ? and sum 1?

my output and label array:
label [
[0 1 1 1 0]
<NDArray 5 @gpu(0)>]
output [
[[ 1.0473623  -1.161462  ]
 [-0.2895019   0.30003032]
 [-0.12274135  0.09577193]
 [ 0.15934949 -0.16305968]
 [ 0.8831042  -0.81932217]]

I used this train code:
#Set Model
net = get_model(name='I3D_resnet50_v1_custom',nclass=2) #dropout=0.9, init_std=0.001) bei resnet50_v1b_custom
net.collect_params().reset_ctx(ctx)

#Epochs
epochs = 10
lr_decay_count = 0
#Training
for epoch in range(epochs):
    tic = time.time()
    train_metric.reset()#accuracy
    train_metric_1.reset()#loss
    train_loss = 0

    # Learning rate decay
    if epoch == lr_decay_epoch[lr_decay_count]:
        trainer.set_learning_rate(trainer.learning_rate*lr_decay)
        lr_decay_count += 1

    # Loop through each batch of training data
    for i, batch in enumerate(train_data):
        # Extract data and label
        data = split_and_load(batch[0], ctx_list=ctx, batch_axis=0)
        label = split_and_load(batch[1], ctx_list=ctx, batch_axis=0)

        # AutoGrad
        with ag.record():
            output = []
            for _, X in enumerate(data):
                X = X.reshape((-1,) + X.shape[2:])
                pred = net(X)
                output.append(pred)
            loss = [loss_fn(yhat, y) for yhat, y in zip(output, label)]

        # Backpropagation
        for l in loss:
            l.backward()

        # Optimize
        trainer.step(batch_size)

        # Update metrics
        train_loss += sum([l.mean().asscalar() for l in loss])
        train_metric.update(label, output)#accuracy
        train_metric_1.update(label, output)#loss
        print("label",label)
        print("output",output)
        if i == 100:
          break

    name, acc = train_metric.get()#accuracy
    name, Loss = train_metric_1.get()#loss

    # Update history and print metrics
    train_history.update([acc])#accuracy
    train_history_1.update([(train_loss/(i+1))])#loss

    print('[Epoch %d] train=%f loss=%f time: %f' %
        (epoch, acc, train_loss / (i+1), time.time()-tic))

    # Export metrics to comet
    #experiment.log_metric("accuracy", acc, epoch=epoch)
    #experiment.log_metric("epoch", epoch)
    #experiment.log_metric("train_loss", train_loss, epoch=epoch)

# We can plot the metric scores with:
train_history.plot()
train_history_1.plot()

#Parameter/Model speichern
net.save_parameters("test.params")

**Comment by bryanyzhu** [comment]
Glad to know that the model is working well on your own dataset. 

In terms of the output, you can see that the last layer of our network is a [fc layer](https://github.com/dmlc/gluon-cv/blob/master/gluoncv/model_zoo/action_recognition/i3d_resnet.py#L534), there is no sigmoid or softmax operation after it. So it is not supposed to between 0 and 1, and the sum is not supposed to be 1. 

But when we do the [loss computation](https://github.com/dmlc/gluon-cv/blob/master/scripts/action-recognition/train_recognizer.py#L454), we use [SoftmaxCrossEntropyLoss](https://github.com/dmlc/gluon-cv/blob/9eeeb175bd083f9ab22063ed7ec42bfd6ebfd41e/gluoncv/loss.py#L268). This function will first apply softmax operation to the network output, making it into a probability, and then computing the cross-entropy loss. 

So to sum up, we are not doing the softmax operation in network forward, but in loss computation. If you need probability score, you can do a softmax after network's output by yourself. Hope this clarifies.

**Comment by yh80ywol** [comment]
First, thanks for your amazing support.

Yes, i can follow your explanation. 
But am i right, that the "loss" / "train_loss" value in code above is already Cross-entropy loss?

And my Trainresults are like this:
[Epoch 0] train=0.488889 loss=0.687800 time: 2660.778338
[Epoch 1] train=0.733333 loss=0.488430 time: 2543.430673
[Epoch 2] train=0.922222 loss=0.196962 time: 2460.234062
[Epoch 3] train=0.966667 loss=0.093875 time: 2444.105297

Is it normal that the train_acc increases so fast? i mean the epochs last very long (2500seconds) but in just 3 epochs the acc gets nearly to 1.  (200 Train videos, each last around 10seconds)
Or should i just output another parameter than epoch-to-acc? iterations-to-acc or something?

**Comment by bryanyzhu** [comment]
Yes, you are right, the loss in the code above is cross-entropy loss. 

In terms of results, this is normal. There are several reasons:
1. When you call `I3D_resnet50_v1_custom`, it is loading a Kinetics400 pretrained model, which is well initialized. 
2. Your problem seems like binary classification, it is relatively easy if your data is clean.
3. Your dataset is small, but `I3D_resnet50_v1_custom` is a rather big model. So in this case, it either works very well or doesn't work at all, depending on your data quality. 

Besides, when I use this model to finetune on UCF101, I also get fast convergence, it gets 80% accuracy in epoch 3. You can check this [log](https://raw.githubusercontent.com/dmlc/web-data/master/gluoncv/logs/action_recognition/ucf101/i3d_resnet50_v1_ucf101_kinetics400.log) for more details. So I think you are fine.

**Comment by yh80ywol** [comment]
Dont know what to say, your awesome.
Thank you very much for your help!
Your explanations are great and relative easy to follow! I really appreciate that.
